src/blockdev.rs

```diff
@@ -993,11 +993,12 @@ pub fn get_blkdev_deps_recursing(device: &Path) -> Result<Vec<PathBuf>> {
     Ok(ret)
 }
 
-fn reread_partition_table(file: &mut File) -> Result<()> {
+fn reread_partition_table(file: &mut File, retry: bool) -> Result<()> {
     let fd = file.as_raw_fd();
     // Reread sometimes fails inexplicably.  Retry several times before
     // giving up.
-    for retries in (0..20).rev() {
+    let max_retries = if retry { 20 } else { 1 };
```

Comment:
This is actually `max_tries`.  (But the name of the `retries` variable is correct, because it contains the number of retries remaining.)